Automattic/wp-calypso
# Stats: Apply module toggling API along with modules layout changing

Related to #77718 

## Proposed Changes

* Add Redux and data layer with modules toggling API to `StatsNavigation` and the `Traffic` page.
* Rearrange module list layout with hidden modules on PC, tablet, and mobile.
* ~Refactor the module toggling panel to adapt to `SectionNav` appropriately.~ (I reverted the commit to make this round of  review easier)

## Testing Instructions



* Spin this change with the live branch link.
* Navigate to Stats > `Traffic` page.
* Click the module toggling panel launch button.
* Toggle modules on the `Popover`.
* Ensure the module list layout changes according to the toggled modules' status.
* Reload the page.
* Ensure the module toggling changes are stored.

<img width="1307" alt="截圖 2023-06-07 上午6 51 33" src="https://github.com/Automattic/wp-calypso/assets/6869813/5ea71f5a-0d81-4de9-8d52-2cb3cfa62bec">

<img width="1468" alt="截圖 2023-06-07 上午7 11 27" src="https://github.com/Automattic/wp-calypso/assets/6869813/38f3a2ee-8f61-4ebf-a9c2-acf4ba61f17e">

## Pre-merge Checklist
